### PR TITLE
New Modules: AWS Network Firewall - rule groups

### PR DESCRIPTION
**Pull request by tremble**
Sorta-Depends-On: https://github.com/ansible-collections/community.aws/pull/974

##### SUMMARY

Two new modules for AWS Network Firewall rule groups.  This first iteration will only support *stateful* rule groups.

networkfirewall_rule_group.py
networkfirewall_rule_group_info.py

ToDo:
- [x] Initial modules
- [x] Return Value documentation
- [x] Integration Tests
- [x] CI Permissions

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

plugins/module_utils/networkfirewall.py
plugins/modules/networkfirewall_rule_group.py
plugins/modules/networkfirewall_rule_group_info.py

##### ADDITIONAL INFORMATION

Note: It's a deliberate choice not to support creation of stateless rules *initially*.  I want to get some of the initial framework in place so that the Policy and Firewall pieces can be built out while waiting on reviews.

### CODE REVIEW

**Comment by gravesm** [comment]
recheck

**Comment by gravesm** [comment]
@tremble We needed to temporarily back out the terminator policy changes. I'll try to sort those out tomorrow.

**Comment by tremble** [comment]
> @tremble We needed to temporarily back out the terminator policy changes. I'll try to sort those out tomorrow.

Sure.  Having problems with botocore versions?  The terminator changes need at least botocore 1.19.20.

FWIW this PR is only for the "Rule Groups", which have no direct costs associated with them.

File: plugins/modules/networkfirewall_rule_group.py [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+    type: str
+  rule_type:
+    description:
+      - Indicates whether the rule group is stateless or stateful.
+      - Stateless rulesets are currently not supported.
+      - Required if I(name) is set.
+    required: false
+    aliases: ['type' ]
+    choices: ['stateful']
+#    choices: ['stateful', 'stateless']
```
Comment (alinabuzachis) [inline plugins/modules/networkfirewall_rule_group.py:38]: Same here.

File: plugins/modules/networkfirewall_rule_group.py [hunk 2]
```diff
@@ -0,0 +1,818 @@
...
+        destination_port=dict(type='str', required=True),
+        sid=dict(type='int', required=True),
+        rule_options=dict(type='dict', required=False),
+    )
+
+    argument_spec = dict(
+        arn=dict(type='str', required=False),
+        name=dict(type='str', required=False),
+        rule_type=dict(type='str', required=False, aliases=['type'], choices=['stateful']),
+        # rule_type=dict(type='str', required=True, aliases=['type'], choices=['stateless', 'stateful']),
```
Comment (alinabuzachis) [inline plugins/modules/networkfirewall_rule_group.py:735]: Is this the correct one for `rule_type`?

File: plugins/modules/networkfirewall_rule_group.py (hunk 1 above)
Comment (tremble) [reply plugins/modules/networkfirewall_rule_group.py:38]: (and yes)

**Comment by gravesm** [comment]
Sorry for the delay. This should be good from the terminator side of things.

**Comment by tremble** [comment]
@goneri Am I missing something silly, this isn't picking up the code from #974 which I added a `Depends-On: ` for.

**Comment by goneri** [comment]
> @goneri Am I missing something silly, this isn't picking up the code from #974 which I added a `Depends-On: ` for.

AFAIK, It needs to be at the top of the file.

**Comment by tremble** [comment]
> AFAIK, It needs to be at the top of the file.

That did it, thanks @goneri

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**